src/views/AddItem.jsx

```diff
@@ -3,30 +3,50 @@ import { Button, FormInput } from '../components';
 import { Link } from 'react-router-dom';
 import { addItem } from '../api';
 
-export function AddItem({ listToken }) {
+export function AddItem({ listToken, itemList }) {
 	const [formFields, setFormFields] = useState({
 		itemName: '',
 		daysUntilNextPurchase: 7,
 	});
 
-	const [message, setMessage] = useState('');
+	
+	const [message, setMessage] = useState(''); 
 
 	const { itemName, daysUntilNextPurchase } = formFields;
 
 	const handleSubmit = async (e) => {
 		e.preventDefault();
 
+		const itemNamesNoChar = itemList.map((item) =>
+			item.name.toLowerCase().replace(/[^a-zA-Z0-9 ]/g, ''),
+		); // Item Names without special characters
+
 		try {
-			await addItem(listToken, {
-				itemName,
-				daysUntilNextPurchase,
-			});
-			setMessage(`${itemName} successfully saved to the database`);
+			if (itemName.replace(/\s/g, '').length === 0) {
```

Comment:
what exception does this error message print when the user uses the app? Tried to find it out but couldn't understand.

Reply:
Not sure which error message you mean but the one related to "await" comes from a previous task that checks if the user has a token, he/she will be able to start saving items, since there is already an existing shopping list.
The first condition checks if the user enters an empty input.